### PR TITLE
[#93] Fixed Github actions crashes on the build step

### LINKED ISSUE #93: Github actions crashes on the build step
**OS:** Windows 10

**Steps to reproduce:**

1. Push code on the PR #91 or #90

**Expected result:**

The github action runner should pass on the build step.

**Actual result:**

Return of a package not found error by the github runner.

**Visual proof:**

### PR DESCRIPTION
**Pull request by ButlerJAL**
# General info

Fixed Github actions crashes on the build step with separate embedded database configuration for testing

**Issue number**: #93 

**Task description**: 

 - Updated src/main/resources/application.properties
 - Created src/test/resources/application.properties
 - Delete src/main/resources/data.sql (clean up)
 - Added H2 Database dependency
 - Updated .github/workflows/build.yml

# Testing

**Test coverage**:
n/a

# Additional notes

**Note to reviewers**:
I changed the production database name to 'switchboard' instead of 'testdb'. Once we get a production database running remotely, it would be wise not to include such properties in a public repo.

**To do before merging**:
n/a

### CODE REVIEW

File: src/main/resources/application.properties
Comment (felixlapierre) [inline src/main/resources/application.properties]: I feel like it would be ideal if the database name (and password) were easily configurable locally. anyone have ideas on how to do that? maybe a third `application.properties` that's gitignored so we can override there? or just IDE runtime configuration overrides?

**Review by felixlapierre — DISMISSED** [review]
Looks good to me - super excited to finally have code coverage working on the backend again. My one comment is more of a discussion starter, not a suggested change.

File: src/main/resources/application.properties
Comment (ButlerJAL) [reply src/main/resources/application.properties]: Yes, we should all have some `.properties` file that is gitignored which includes credentials.

Comment (ButlerJAL) [reply src/main/resources/application.properties]: Actually, on second thought, we can simply use the same `applications.properties` with TODO comment <INSERT CREDENTIALS> placeholder for the username and password property values

**Review by mohhef — DISMISSED** [review]
Great addition.

File: src/main/resources/application.properties
Comment (ButlerJAL) [reply src/main/resources/application.properties]: @felixlapierre see last commit

Comment (ButlerJAL) [reply src/main/resources/application.properties]: We could explore better options down the road, but for now I think this works

Comment (felixlapierre) [reply src/main/resources/application.properties]: The problem with using TODO is that when we fill in the value locally it will always appear as a change that is ready to be committed. So you have to be careful not to accidentally commit your local username & password which is very annoying.

Comment (felixlapierre) [reply src/main/resources/application.properties]: IMO we can leave it as default values for now, which will work for most people's setups, and come up with a better solution in a later PR.

Comment (ButlerJAL) [reply src/main/resources/application.properties]: @felixlapierre done

**Review by felixlapierre — APPROVED** [review]
re-approving since my previous review was made stale, still LGTM

**Review by mohhef — APPROVED** [review]
Re-approving after verifying commits pushed after my initial approval.

**Pull request merged**
